src/main/scala/com/actian/spark_vector/sql/SparkSqlTable.scala

```diff
@@ -0,0 +1,30 @@
+package com.actian.spark_vector.sql
+
+import java.util.concurrent.atomic.AtomicLong
+import org.apache.spark.sql.DataFrame
+
+sealed trait SparkSqlTable {
+  def tableName: String
+  def quotedName = sparkQuote(tableName)
+  def close(): Unit
+}
+
+case class HiveTable(override val tableName: String) extends SparkSqlTable {
+  override def close(): Unit = { }
+}
+
+class TempTable private (override val tableName: String, df: DataFrame) extends SparkSqlTable {
```

Comment:
For TempTables, we want to control how such objects are created (we want to be sure that the register method is called immediately after creation).

On that note, to guarantee that register isn't called multiple times, make it private.